src/SdkCommon/ClientRuntime/ClientRuntime/RetryAfterDelegatingHandler.cs

```diff
@@ -44,35 +53,72 @@ public RetryAfterDelegatingHandler(DelegatingHandler innerHandler)
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
             CancellationToken cancellationToken)
         {
+            HttpResponseMessage previousResponseMessage = null;
             do {
-                var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                
-                // if they send back a 429
-                if( response.StatusCode == ((HttpStatusCode)429) ) {
+                HttpResponseMessage response = null;
 
-                    // and there is a retry-after header
-                    if (response.Headers.Contains("Retry-After"))
+                try
+                {
+                    response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+                    // if they send back a 429
+                    if (response.StatusCode == (HttpStatusCode)429)
                     {
-                        try {
-                            // and we get a number of seconds from the header
-                            string retryValue = response.Headers.GetValues("Retry-After").FirstOrDefault();
-                            var retryAfter = int.Parse(retryValue, CultureInfo.InvariantCulture);
-                            
-                            // wait for that duration
-                            await Task.Delay(TimeSpan.FromSeconds(retryAfter), cancellationToken);
+                        // and there is a retry-after header
+                        if (response.Headers.Contains("Retry-After"))
```

Comment:
Why not capture the content for all the attempts? Could there be useful information in the previous responses that we are discarding?

Reply:
In the end we can only return 1 response and the latest response has the most up to date information.

Reply:
I was wondering if we could concatenate all the responses and set the concatenated response as the one we send back -- but I am fine with just using the last one